sqlair.go

```diff
@@ -279,6 +287,9 @@ func (q *Query) Iter() *Iterator {
 		sqlstmt = q.tx.sqltx.Stmt(q.sqlstmt)
 		close = sqlstmt.Close
 	}
+	if q.isTemp {
+		close = sqlstmt.Close
+	}
```

Comment:
I think a finalizer would be a bit too heavyweight here. We know exactly when we can close the `sql.Stmt` so we should just close it there.
If we have a LRU cache or some other way of keeping these `sql.Stmt` values then we should close them on eviction from the cache. If they are in the cache then the finalizer on them will never run. We can still have a finalizer on the `sqlair.Statement` because that is not in the cache itself and we can close *all* the `sql.Stmt` that correspond to the different versions of the statement.

Reply:
Yeah you are right, we know exactly when to close the statements both with the current implementation and with the LRU cache so finalizers are not necessary.

But what happens if the user creates a `Query` object `q`, never calls `Get` and then `q` goes out of scope. The temporary statement would never be closed, right?